scripts/start.js

```diff
@@ -66,62 +69,89 @@ function clearConsole() {
-  console.log('Compiling...');
-});
-compiler.plugin('done', function (stats) {
-  clearConsole();
-  var hasErrors = stats.hasErrors();
-  var hasWarnings = stats.hasWarnings();
-  if (!hasErrors && !hasWarnings) {
-    console.log(chalk.green('Compiled successfully!'));
-    console.log();
-    console.log('The app is running at http://localhost:3000/');
-    console.log();
-    return;
-  }
+detect(PORT, function(error, _port) {
 
-  var json = stats.toJson();
-  var formattedErrors = json.errors.map(message =>
-    'Error in ' + formatMessage(message)
-  );
-  var formattedWarnings = json.warnings.map(message =>
-    'Warning in ' + formatMessage(message)
-  );
+  if (PORT !== _port) {
+    var rl = readline.createInterface({
+      input: process.stdin,
+      output: process.stdout
+    });
 
-  if (hasErrors) {
-    console.log(chalk.red('Failed to compile.'));
-    console.log();
-    if (formattedErrors.some(isLikelyASyntaxError)) {
-      // If there are any syntax errors, show just them.
-      // This prevents a confusing ESLint parsing error
-      // preceding a much more useful Babel syntax error.
-      formattedErrors = formattedErrors.filter(isLikelyASyntaxError);
-    }
-    formattedErrors.forEach(message => {
-      console.log(message);
-      console.log();
+    rl.question('Something is already running at http://localhost:' + PORT + '. Would you like to run the app at another port instead? [Y/n] ', function(answer){
+      if(answer === 'Y') {
+        PORT = _port;
+        config.entry = config.entry.map(function(c){ return c.replace(/(\d+)/g, PORT) }); // Replace the port in webpack config
+        compiler = webpack(config, handleCompile);
+        setupCompiler();
+        runDevServer();
+      }
+      rl.close();
     });
-    // If errors exist, ignore warnings.
-    return;
+  } else {
+    runDevServer();
+    setupCompiler();
   }
+});
 
-  if (hasWarnings) {
-    console.log(chalk.yellow('Compiled with warnings.'));
-    console.log();
-    formattedWarnings.forEach(message => {
-      console.log(message);
+function setupCompiler() {
+  compiler.plugin('invalid', function () {
+    clearConsole();
+    console.log('Compiling...');
+  });
+
+  compiler.plugin('done', function (stats) {
+    clearConsole();
+    var hasErrors = stats.hasErrors();
+    var hasWarnings = stats.hasWarnings();
+    if (!hasErrors && !hasWarnings) {
+      console.log(chalk.green('Compiled successfully!'));
       console.log();
-    });
+      console.log('The app is running at http://localhost:' + PORT + '/');
+      console.log();
+      return;
+    }
 
-    console.log('You may use special comments to disable some warnings.');
-    console.log('Use ' + chalk.yellow('// eslint-disable-next-line') + ' to ignore the next line.');
-    console.log('Use ' + chalk.yellow('/* eslint-disable */') + ' to ignore all warnings in a file.');
-  }
-});
+    var json = stats.toJson();
+    var formattedErrors = json.errors.map(message =>
```

Comment:
Is there a reason we're using arrow functions and standard functions in the same file?  I don't mind either syntax but I think consistency is a good idea. That is, unless we need the syntax difference for the different ways they handle things, but if it's just arbitrary I'd prefer we just stuck with one :)